packages/react-instantsearch-hooks-web/src/ui/__tests__/HierarchicalMenu.test.tsx

```diff
@@ -412,4 +417,23 @@ describe('HierarchicalMenu', () => {
       'Some custom title'
     );
   });
+
+  test('renders with translations', () => {
+    const props = createProps();
+    const { getByRole } = render(
+      <HierarchicalMenu
+        {...props}
+        showMore
+        translations={{
+          showMoreButtonText({ isShowingMore }) {
+            return isShowingMore ? 'Show less brands' : 'Show more brands';
+          },
+        }}
+      />
+    );
+
+    expect(
+      getByRole('button', { name: 'Show more brands' })
+    ).toBeInTheDocument();
+  });
```

Comment:
Should we re-render with `showMore=false` to test the other side of the ternary?